File: GXpert/GXpert.Web/Modules/Settings/Sms/SmsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Settings.Columns;

[ColumnsScript("Settings.Sms")]
[BasedOnRow(typeof(SmsRow), CheckNames = true)]
public class SmsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string ApiKey { get; set; }
    public string Sender { get; set; }
    public string EntityId { get; set; }
    public string TemplateId { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Settings/SmsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SmsRow } from "./SmsRow";

export interface SmsColumns {
    Id: Column<SmsRow>;
    ApiKey: Column<SmsRow>;
    Sender: Column<SmsRow>;
    EntityId: Column<SmsRow>;
    TemplateId: Column<SmsRow>;
    InsertDate: Column<SmsRow>;
    InsertUserId: Column<SmsRow>;
    UpdateDate: Column<SmsRow>;
    UpdateUserId: Column<SmsRow>;
    IsActive: Column<SmsRow>;
}

export class SmsColumns extends ColumnsBase<SmsRow> {
    static readonly columnsKey = 'Settings.Sms';
    static readonly Fields = fieldsProxy<SmsColumns>();
}